### ISSUE TITLE
Video classification TensorFlow Lite model(i3d) with 5D input on android 

### ISSUE THREAD

**Issue by kishorkuttan** [tensorflow/tensorflow#49612]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Vivo Y91i
- TensorFlow installed from (source or binary): binary
- TensorFlow version: v2.5.0
- Python version: 3.7
- Installed using virtualenv? pip? conda?: pip
- Bazel version (if compiling from source): NIL
- GCC/Compiler version (if compiling from source): NIL
- CUDA/cuDNN version: CUDA 11.0
- GPU model and memory: 11 GB

**Describe the problem**
Need to run a sign language video classification model(i3d) on android, that takes 5 dimension ( 1 3 64 224 224),https://github.com/tensorflow/examples/tree/master/lite/examples/image_classification/android , but failes. But runs on PC with python3.7
```bash
import random
import re
import os
import tempfile
import ssl
import cv2
import scipy.special
import numpy as np
import pickle as pkl
import math
import copy
import tensorflow as tf
# Some modules to display an animation using imageio.
import imageio
from IPython import display

from urllib import request  # requires python3
# Utilities to open video files using CV2
def crop_center_square(frame):
  y, x = frame.shape[0:2]
  min_dim = min(y, x)
  start_x = (x // 2) - (min_dim // 2)
  start_y = (y // 2) - (min_dim // 2)
  return frame[start_y:start_y+min_dim,start_x:start_x+min_dim]

def load_label():
  with open("labels.txt", 'r') as f:
    labels = [line.strip() for line in f.readlines()]
    return labels

# load the label
word_data = load_label(
    
  )
# load the tensorflow lite run time

interpreter = tf.lite.Interpreter(model_path="onnx_lite64.tflite")
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

# resize
resize = (224,224)
path = "test.mp4"
# read the video
cap = cv2.VideoCapture(path)

# Get the Default resolutions
frame_width = int(cap.get(3))
frame_height = int(cap.get(4))

# Define the codec and filename.
out = cv2.VideoWriter('output.avi',cv2.VideoWriter_fourcc('M','J','P','G'), 5, (frame_width,frame_height))
frames = []

in_frames = 65
counter = 1


try:
  while True:
   
    ret, frame_old = cap.read()
    if not ret:
      break
    frame_new = copy.deepcopy(frame_old)
    frame = crop_center_square(frame_old)

    frame = cv2.resize(frame, resize)
    frame = frame[:, :, [2, 1, 0]] 
    
    frames.append(frame)
    # check the counter is not zero and append the frames
    if counter!=0:

      if counter % in_frames == 0:
      

        del frames[:]

    else:

     
      frames.append(frame)
     
     
    

    
    res = np.array(frames) / 255.0
    print(res.shape)
    value = res.shape[0] 
    counter+=1
   
  

    # Remove the frame with zero clip
    if value !=0 and value == 64:
      #print(res.shape)
      model_input = tf.constant(res, dtype=tf.float32)[tf.newaxis, ...]
      print(model_input.shape)
      inp = tf.transpose(model_input, perm=[0, 4, 1, 2,3])
      print(inp.shape)
      interpreter.set_tensor(input_details[0]['index'], inp)
      interpreter.invoke()
      outputs = interpreter.get_tensor(output_details[0]['index'])
      topk=1
      result = []
      num_clips =1
      num_detections = 2000
      for i in range(num_detections):
        res = outputs[0][i]
        result.append(res)
      #print(result)
      res = []
      res.append(result)
      
      raw_scores = np.array(res)

      prob_scores = scipy.special.softmax(raw_scores, axis=1)
      prob_sorted = np.sort(prob_scores, axis=1)[:, ::-1]
      pred_sorted = np.argsort(prob_scores, axis=1)[:, ::-1]
     



      word_topk = None
      for k in range(topk):
          for i, p in enumerate(pred_sorted[:, k]):
              if str(word_data[p])!="want":
                  print(str(word_data[p]))
              
                  word_topk= word_data[p]
      prob_topk = prob_sorted[:, :topk].transpose()
      print("Predicted signs:")
      print(word_topk)
      print(type(prob_topk))


      cv2.putText(frame_old, str(word_topk),(30,50),cv2.FONT_HERSHEY_SIMPLEX,1,(255,255,0),2,cv2.LINE_AA)
      out.write(frame_old)
      cv2.imshow("Result",frame_old)

  # All the results have been drawn on the frame, so it's time to display it.
    cv2.imshow('rgb', frame_new)


    if cv2.waitKey(1) == ord('q'):
      break


  #word_data = load_label()

finally:
  cap.release()

```
## Error on Android studio
![java](https://user-images.githubusercontent.com/48623612/119530465-a5dbe200-bda0-11eb-9a1e-5331b514fbd4.png)

**Provide the exact sequence of commands / steps that you executed before running into the problem**
The model was converted from PyTorch > TF .> TFLite. The code throws error while running on android studio.

![input](https://user-images.githubusercontent.com/48623612/119531294-68c41f80-bda1-11eb-8ae2-faaee6fc89e7.png)
![output](https://user-images.githubusercontent.com/48623612/119531305-6b267980-bda1-11eb-9852-1c597d11c502.png)

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by abattery** [comment]
Please consider using the fixed input tensor shapes and FlexAddV2 can be gone if there are only fixed shapes in the input tensors of the add op.

**Comment by kishorkuttan** [comment]
But how can we input (1 3 64 224 224) instead of (1 224 224 3) in the Java script.

**Comment by abattery** [comment]
For the JavaScript question, could you upload a new post about that in order to keep each issue focused?

**Comment by kishorkuttan** [comment]
![119530465-a5dbe200-bda0-11eb-9a1e-5331b514fbd4](https://user-images.githubusercontent.com/48623612/119594686-e156ca80-bdf9-11eb-9d6f-3331143de423.png)
This is the error. I think it is due to the input size. In opencv we can concat frames and convert it to a numpy array to get the size as (64 224 224 3) then transpose it to (1 3 224 224). No idea on how it can be done with Java.

**Comment by kishorkuttan** [comment]
No issue on python3.

**Comment by kishorkuttan** [comment]
> For the JavaScript question, could you upload a new post about that in order to keep each issue focused?

sorry for the typo, it's java

**Comment by kishorkuttan** [comment]
Any updates

**Comment by kishorkuttan** [comment]
I just tried many ways. The possibile solution is bitmat to bytebuffer conversion. After many trial and error. It runs but the labels are not accurate

**Comment by kishorkuttan** [comment]
![Uploading IMG_20210527_223254.jpg…]()

**Comment by kishorkuttan** [comment]
I think the 5 D input is the issue. How can put the float values in the bytebuffer without messing up the input

**Comment by kishorkuttan** [comment]
![IMG_20210527_223254](https://user-images.githubusercontent.com/48623612/119868578-d789b000-bf3c-11eb-92cb-3070ad7f35f0.jpg)

**Comment by kishorkuttan** [comment]
Has anyone successfully run  5D video tensors as input in android java
